ledger/complete/wal/fadvise.go

```diff
@@ -0,0 +1,44 @@
+package wal
+
+import (
+	"runtime"
+
+	"golang.org/x/sys/unix"
+)
+
+// fadviseNoLinuxPageCache advises Linux to evict
+// a file from Linux page cache.  This calls unix.Fadvise which
+// in turn calls posix_fadvise with POSIX_FADV_DONTNEED.
+// If GOOS != "linux" then this function does nothing.
+// CAUTION: If fsync=true, this will call unix.Fsync which
+// can cause performance hit especially when used inside a loop.
+func fadviseNoLinuxPageCache(fd uintptr, fsync bool) error {
+	return fadviseSegmentNoLinuxPageCache(fd, 0, 0, fsync)
+}
+
+// fadviseSegmentNoLinuxPageCache advises Linux to evict the
+// file segment from Linux page cache.  This calls unix.Fadvise
+// which in turn calls posix_fadvise with POSIX_FADV_DONTNEED.
+// If GOOS != "linux" then this function does nothing.
+// CAUTION: If fsync=true, this will call unix.Fsync which
+// can cause performance hit especially when used inside a loop.
+func fadviseSegmentNoLinuxPageCache(fd uintptr, off, len int64, fsync bool) error {
+	if runtime.GOOS != "linux" {
+		return nil
+	}
```

Comment:
Maybe just use a build tag instead to move this check to compile-time?

Reply:
Great suggestion! 👍 Will do.

Reply:
Done in commit 15cdac5fe04f3997141e1ddc8a8e02f86987d0f0.